src/main/java/com/scaleunlimited/flinkcrawler/tools/CrawlTool.java

```diff
@@ -82,6 +112,9 @@ public void setMaxContentSize(int maxContentSize) {
 		
 		@Option(name = "-commoncrawl", usage = "crawl id for CommonCrawl.org dataset", required = false)
 	    public void setCommonCrawlId(String commonCrawlId) {
+            if (getUserAgent() != null) {
+                throw new RuntimeException("user agent not used in common crawl mode");
```

Comment:
As as above, I think you have to validate the options after they've all been parsed, to avoid order dependency issues.

Reply:
I think it works as long as the check is in both places.